sdk/mobile-center-distribute/src/test/java/com/microsoft/azure/mobile/distribute/DistributeMandatoryDownloadTest.java

```diff
@@ -359,4 +360,103 @@ private void verifyWithInvalidOrMissingCachedJson() throws Exception {
         verifyStatic();
         PreferencesStorage.remove(PREFERENCE_KEY_DOWNLOAD_STATE);
     }
+
+    @Test
+    public void newOptionalUpdateWhileInstallingMandatory() throws Exception {
+
+        /* Mock mandatory download and showing install U.I. */
+        waitDownloadTask();
+        completeDownload();
+        mockSuccessCursor();
+        Intent installIntent = mockInstallIntent();
+        waitCheckDownloadTask();
+        waitCheckDownloadTask();
+        verify(mContext).startActivity(installIntent);
+        verifyStatic();
+        PreferencesStorage.putInt(PREFERENCE_KEY_DOWNLOAD_STATE, DOWNLOAD_STATE_AVAILABLE);
+        verifyStatic();
+        PreferencesStorage.putInt(PREFERENCE_KEY_DOWNLOAD_STATE, DOWNLOAD_STATE_INSTALLING);
+        verifyNoMoreInteractions(mNotificationManager);
+
+        /* Restart will restore install. */
+        restartProcessAndSdk();
+        Distribute.getInstance().onActivityResumed(mActivity);
+        waitCheckDownloadTask();
+        verify(mContext, times(2)).startActivity(installIntent);
+        verifyStatic(times(2));
+        PreferencesStorage.putInt(PREFERENCE_KEY_DOWNLOAD_STATE, DOWNLOAD_STATE_INSTALLING);
+        Distribute.getInstance().onActivityPaused(mActivity);
+
+        /* Change what the next detected release will be: a more recent mandatory one. */
+        ReleaseDetails releaseDetails = mock(ReleaseDetails.class);
+        when(releaseDetails.getId()).thenReturn(5);
+        when(releaseDetails.getVersion()).thenReturn(8);
+        when(releaseDetails.getDownloadUrl()).thenReturn(mDownloadUrl);
+        when(releaseDetails.isMandatoryUpdate()).thenReturn(false);
```

Comment:
No too difficult to use that jUnit feature in those after download tests, already tried and it was a mess ;)